src/routes/workouts-route.php

```diff
@@ -145,68 +95,44 @@ public function get_workouts() {
-			'orphaned'    => $request['orphaned'],
-		];
+		$workouts_data = $request->get_json_params();
+
+		$free_workouts_data                  = [];
+		$free_workouts_data['configuration'] = $workouts_data['configuration'];
 
-		foreach ( $value as $workout => $data ) {
-			if ( isset( $data['indexablesByStep'] ) && \is_array( $data['indexablesByStep'] ) ) {
-				foreach ( $data['indexablesByStep'] as $step => $indexables ) {
-					if ( $step === 'removed' ) {
-						continue;
-					}
-					$value[ $workout ]['indexablesByStep'][ $step ] = \wp_list_pluck( $indexables, 'id' );
-				}
-			}
-		}
+		$result = $this->options_helper->set( 'workouts_data', $free_workouts_data );
+
+		/**
+		 * Filter: 'Yoast\WP\SEO\workouts_route_save' - Allows the add-ons to save the options data in their own options.
+		 *
+		 * @api mixed|null $result The result of the previous saving operation.
+		 *
+		 * @param array $workouts_data The full set of workouts option data to save.
+		 */
+		$result = \apply_filters( 'Yoast\WP\SEO\workouts_route_save', $result, $workouts_data );
 
 		return new WP_REST_Response(
-			[ 'json' => \YoastSEO()->helpers->options->set( 'workouts', $value ) ]
+			[ 'json' => $result ]
 		);
 	}
 
 	/**
-	 * Validates the cornerstone attribute
-	 *
-	 * @param array $workout The cornerstone workout.
-	 * @return bool If the payload is valid or not.
-	 */
-	public function cornerstone_is_allowed( $workout ) {
-		return $this->is_allowed( $workout, self::ALLOWED_CORNERSTONE_STEPS );
-	}
-
-	/**
-	 * Validates the orphaned attribute
+	 * Gets the args for all the registered workouts.
 	 *
-	 * @param array $workout The orphaned workout.
-	 * @return bool If the payload is valid or not.
+	 * @return array
 	 */
-	public function orphaned_is_allowed( $workout ) {
-		return $this->is_allowed( $workout, self::ALLOWED_ORPHANED_STEPS );
-	}
+	private function get_workouts_routes_args() {
+		$args_array = [
+			'configuration' => [
+				'required' => true,
+			],
+		];
 
-	/**
-	 * Validates a workout.
-	 *
-	 * @param array $workout       The workout.
-	 * @param array $allowed_steps The allowed steps for this workout.
-	 * @return bool If the payload is valid or not.
-	 */
-	public function is_allowed( $workout, $allowed_steps ) {
-		// Only 2 properties are allowed, the below validated finishedSteps property.
-		if ( \count( $workout ) !== 2 ) {
-			return false;
-		}
-
-		if ( isset( $workout['finishedSteps'] ) && \is_array( $workout['finishedSteps'] ) ) {
-			foreach ( $workout['finishedSteps'] as $step ) {
-				if ( ! \in_array( $step, $allowed_steps, true ) ) {
-					return false;
-				}
-			}
-			return true;
-		}
-		return false;
+		/**
+		 * Filter: 'Yoast\WP\SEO\workouts_route_args' - Allows the add-ons add their own arguments to the route registration.
+		 *
+		 * @api array $args_array The array of arguments for the route registration.
+		 */
+		return \apply_filters( 'Yoast\WP\SEO\workouts_route_args', $args_array );
```

Comment:
See comment above about namespace in the hook name.

Reply:
Same :)